accord-core/src/test/java/accord/utils/LoggingRandomSource.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package accord.utils;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This class is helpful for finding non-deterministic behavior as it tracks every call to {@link RandomSource} so that
+ * different runs can be compared to find where they differ.  The most common way to use this is to replace the tests
+ * {@link RandomSource} with this class and run it against 2 or more environments (that are experiencing a different outcome),
+ * then compare the logs to find where the first difference is (and the previous access is likely where the non-deterministic fault is).
+ */
+@SuppressWarnings("unused")
+public class LoggingRandomSource implements RandomSource
```

Comment:
I wrote this to help find where the histories were drifting, and was able to find `oneOf` was at fault.  Rather than implementing w/e we find something non-deterministic can use this to help find it.